usaspending_api/api_contracts/contracts/v2/search/spending_by_award.md

```diff
@@ -232,6 +233,10 @@ List of table columns
 + `exclude`: `3333` (optional, array[string])
 
 ### TASCodeObject (object)
++ `require`: [[`091`]] (optional, array[array[string]])
```

Comment:
`array[array[string]]` ?

Reply:
Is that inappropriate syntax? The endpoint DOES use a 2D array.

Reply:
Oh interesting. Can you share a request JSON example?

Reply:
```
"tas_codes": {
        	"require": [["091"]],
        	"exclude": [["091","091-0800"]]
        },
```

Reply:
Because the relationship between Agency, Federal Account, and TAS are arbitrary (at least to the degree that you can't know for sure what agency a federal account belongs to from the federal account code), the API needs the full ancestor array in order to link search nodes for the hierarchical logic.

Reply:
Can you show a useful example of leveraging the 2D array? The copy-paste from the API Contract isn't showing why a 2d array is necessary.

Reply:
```
"tas_codes": {
        	"require": [["011"]],
        	"exclude": [["011","011-1082","097-011-2007/2012-1082-000"]]
        },
```
would be an example of where it is nessicary. For whatever reason, the tas loader's rules decided that this TAS was under agency 011, NOT agency 097 like the ata in  the TAS code might imply. Without the path provided (which  is always provided by the filter endpoints themselves, so there's no issue of the javascript  code not having the path), the search logic would have no way to know if  "097-011-2007/2012-1082-000" is a descendent of "011". Even if there was a way,  you are now putting context-specific logic in either the frontend or the search filter, introducing risk of desync with the sql performing the data load.